src/stores/calenderizeHelpers.ts

```diff
@@ -64,10 +63,10 @@ export const calendarizeCourseEvents = () => {
 };
```

Comment:
Also, you don't need to type assert the array if you use functional programming, e.g. map, filter, etc. (those are more TypeScript friendly). Here's an example implementation that seems to get me pretty close (there's issues with the final typing because the CourseEvent extends from the Event class, which the returned object lacks unless you assert it:

```ts
export function calendarizeCourseEvents(currentCourses: Course[]) {
  return currentCourses.map((course) => (
    course.section.meetings
      .filter((meeting) => meeting.time.replace(/\s/g, '') !== 'TBA')
      .map((meeting) => {
        const timeString = meeting.time.replace(/\s/g, '');
        const [, startHrStr, startMinStr, endHrStr, endMinStr, ampm] = timeString.match(
          /(\d{1,2}):(\d{2})-(\d{1,2}):(\d{2})(p?)/
        ) as RegExpMatchArray;

        let startHr = parseInt(startHrStr, 10);
        const startMin = parseInt(startMinStr, 10);
        let endHr = parseInt(endHrStr, 10);
        const endMin = parseInt(endMinStr, 10);

        /**
         * map each day to a boolean indicating whether the course meets on that day
         */
        const dates = days.map(day => meeting.days.includes(day))

        if (ampm === 'p' && endHr !== 12) {
          startHr += 12;
          endHr += 12;
          if (startHr > endHr) startHr -= 12;
        }

        return dates
          .filter((shouldBeInCal) => shouldBeInCal)
          .map((_date, index) => {
            const newEvent = {
              color: course.section.color,
              term: course.term,
              title: course.deptCode + ' ' + course.courseNumber,
              courseTitle: course.courseTitle,
              bldg: meeting.bldg,
              instructors: course.section.instructors,
              sectionCode: course.section.sectionCode,
              sectionType: course.section.sectionType,
              start: new Date(2018, 0, index, startHr, startMin),
              finalExam: course.section.finalExam,
              end: new Date(2018, 0, index, endHr, endMin),
              isCustomEvent: false as const,
            };
            return newEvent;
          });
      }).flat()
  )).flat();
}
```

Reply:
Addressed offline, in summary we can consider this in another PR.